File: problems/2607_make-k-subarray-sums-equal/solution.ts
function makeSubKSumEqual(arr: number[], k: number): number {
    const gcd = (a: number, b: number): number => {
        while (b !== 0) {
            [a, b] = [b, a % b];
        }
        return a;
    };
    const n = arr.length;
    const g = gcd(n, k);
    let total = 0;
    for (let r = 0; r < g; r++) {
        const group: number[] = [];
        for (let i = r; i < n; i += g) group.push(arr[i]);
        group.sort((a, b) => a - b);
        const median = group[Math.floor(group.length / 2)];
        for (const v of group) total += Math.abs(v - median);
    }
    return total;
}

File: problems/2607_make-k-subarray-sums-equal/solution.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {

    public long makeSubKSumEqual(int[] arr, int k) {
        int n = arr.length;
        int g = gcd(n, k);
        long total = 0;
        for (int r = 0; r < g; r++) {
            List<Integer> group = new ArrayList<>();
            for (int i = r; i < n; i += g) group.add(arr[i]);
            Collections.sort(group);
            int median = group.get(group.size() / 2);
            for (int v : group) total += Math.abs((long) v - median);
        }
        return total;
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }
}
